server/templates/template_executor.go

```diff
@@ -42,16 +49,33 @@ func (gte *goTemplateExecutor) Format() string {
 	return "go"
 }
 
+func (gte *goTemplateExecutor) Expression() string {
+	return gte.expression
+}
+
 func (gte *goTemplateExecutor) isPlainText() bool {
 	return len(gte.template.Tree.Root.Nodes) == 1 &&
 		gte.template.Tree.Root.Nodes[0].Type() == parse.NodeText
 }
 
 type jsTemplateExecutor struct {
 	jsFunction func(map[string]interface{}) interface{}
+	transformedExpression string
+}
+
+type asyncJsTemplateExecutor struct {
+	incoming chan events.Event
+	results chan interface{}
+	transformedExpression string
+	loadingError error
 }
 
-func newJsTemplateExecutor(expression string, extraFunctions template.FuncMap) (*jsTemplateExecutor, error) {
+type pooledTemplateExecutor struct {
```

Comment:
is it used anywhere? If not, we shouldn't keep it in the codebase

Reply:
not used. will deleted it